src/Native/Unix/System.Security.Cryptography.Native/pal_evp_cipher.c

```diff
@@ -5,76 +5,75 @@
 #include "pal_evp_cipher.h"
 
 #include <assert.h>
-#include <memory>
 
 #define SUCCESS 1
 #define KEEP_CURRENT_DIRECTION -1
 
-extern "C" EVP_CIPHER_CTX*
+EVP_CIPHER_CTX*
 CryptoNative_EvpCipherCreate(const EVP_CIPHER* type, uint8_t* key, unsigned char* iv, int32_t enc)
 {
     return CryptoNative_EvpCipherCreate2(type, key, 0, 0, iv, enc);
 }
 
-extern "C" EVP_CIPHER_CTX*
+EVP_CIPHER_CTX*
 CryptoNative_EvpCipherCreate2(const EVP_CIPHER* type, uint8_t* key, int32_t keyLength, int32_t effectiveKeyLength, unsigned char* iv, int32_t enc)
 {
-    std::unique_ptr<EVP_CIPHER_CTX> ctx(new (std::nothrow) EVP_CIPHER_CTX);
-    if (ctx == nullptr)
+    EVP_CIPHER_CTX* ctx = (EVP_CIPHER_CTX*)malloc(sizeof(EVP_CIPHER_CTX));
+    if (ctx == NULL)
     {
         // Allocation failed
-        return nullptr;
+        return NULL;
     }
 
-    EVP_CIPHER_CTX_init(ctx.get());
+    EVP_CIPHER_CTX_init(ctx);
 
     // Perform partial initialization so we can set the key lengths
-    int ret = EVP_CipherInit_ex(ctx.get(), type, nullptr, nullptr, nullptr, 0);
+    int ret = EVP_CipherInit_ex(ctx, type, NULL, NULL, NULL, 0);
     if (!ret)
     {
-        return nullptr;
+        return NULL;
```

Comment:
Don't we need to free the ctx in all of these cases where it's been allocated but we return NULL?  Previously it looks like the unique_ptr was handling that.